### PR TITLE
update to 3.34.0

### PR DESCRIPTION
**Pull request by boldorider4**
* [x] bumped version and updated recipe according to latest requirements
* [x] removed unnecessary dependencies
* [x] checked the license file and verified all required license file get distributed with the package

### CODE REVIEW

**Review by ViridianMelody — APPROVED** [review]
Looks good to me. That linter error is really strange, I can't pin down what it's complaining about, besides _maybe_ something to do with the version selector in [line 92](https://github.com/AnacondaRecipes/orange3-feedstock/blob/cfc0dd09152c00b192239fdbe8de2bca05705ad4/recipe/meta.yaml#L92)?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -43,47 +43,40 @@ requirements:
...
+    - joblib >=0.11
     - keyring
     - keyrings.alt
-    - matplotlib-base >=2.2.5
+    - matplotlib-base >=3.2.0
     - networkx
-    - numpy >=1.17.3
+    - numpy >=1.19.5
     - openpyxl
     - opentsne >=0.6.1
```
Comment (ELundby45) [inline recipe/meta.yaml]: https://github.com/biolab/orange3/blob/e0ee124f82b186de0b693dadf98011ccdf449bf2/requirements-core.txt#L23
```suggestion
    - opentsne >=0.6.1,!=0.7.0
```

**Review by ELundby45 — APPROVED** [review]
One minor comment (take it or leave it, up to you). 

Overall LGTM. Nice Work!

**Comment by skupr-anaconda** [comment]
As it's a GUI app I strongly recommend testing from the test channel if it launches correctly and there will be not found visual bugs

File: recipe/meta.yaml (hunk 1 above)
Comment (boldorider4) [reply recipe/meta.yaml]: I'll take it!

**Pull request merged**